PerfNext/app/apis/BenchEngine/parser.js

```diff
@@ -41,24 +41,44 @@ module.exports = function(app) {
     */
 	app.post('/api/benchengine/submit', function(req, res) {
 		var raw_definition = req.rawBody; //Raw XML to be parsed
-
-		parseDefinition(raw_definition, function(generatedOutput){
-
-			var setupScript = parseSetupScript(raw_definition, function(setupScript){
-
+		
+		// Split the received xml file to separate test and baseline scripts
+		var split_definition = raw_definition.split("<?xml");
+		
+		// Add in missing xml tag from previous split
+		var test_raw_definition = "<?xml" + split_definition[1];
+		
+			
+		// If interleave baseline unticked
+		if (split_definition.length < 3) {
+			var baseline_raw_defintion = null;	
+		} else {
+			var baseline_raw_definition = "<?xml" + split_definition[2];
+		}
+		parseDefinition(test_raw_definition, function(generatedOutput){
+			
+			var setupScript = parseSetupScript(test_raw_definition, function(setupScript){
+				
+				parseDefinition(baseline_raw_definition, function(generatedOutputbase){
```

Comment:
Let's move the common code to some function so that we can use that for both baseline and test builds.